### ISSUE TITLE
Best way to handle an edge case where tenant store has duplicate Id, Identifier or ConnectionString 

### ISSUE THREAD

**Issue by mkgn** [Finbuckle/Finbuckle.MultiTenant#533]
This is an edge case which shouldn't happen for the most part. However what happened in my case is that one developer changed the tenants.json file (I am using config file store) for a quick check on a completely different thing, but he copied existing tenant entry, pasted it and changed only the connection string and now there is duplicate id, identifier in the config file.

I think this can be a dangerous situation where multiple things can go wrong. I feel that if a duplicate Id, Identifier or ConnectionString is found we should throw an exception.

I tried to do this by making a custom config store and monitoring changes to the tenants.json file (using IOptionsMonitor<>). Now whenever file changes I load it and validate for duplicates. Unfortunately throwing an error in IOptionsMonitor.OnChange event if duplicates found is not working because of the issue mentioned [here](https://github.com/dotnet/runtime/issues/44381) 

Is there other options available to handle this kind of an edge case?

**Comment by AndrewTriesToCode** [comment]
Hm, that is a tough one. So is the problem more about cleaning up this one conflict or preventing more in the future? This is one good reason to use a database store since it would prevent the conflict in the first place, but that might not be an option for you.

Do you need the config to be dynamic? If you don't use `IOptionsMonitor` you just wouldn't have to worry about it except on app start which you could easily check for. I should probably have something in the library that checks for this when reloading the settings on a configuration changes, but I might hit the same issue you did.-- I'll save this as an enhancement idea.

**Comment by mkgn** [comment]
Yes, having it in Db would solve the problem, but in my case the config file is more suitable for few reasons. Since the library is solid I think any store should have this edge case handled for completeness. Anyway what I did is kind of a hack outside of the custom tenantstore I created.

I have a TenantDisabled middleware that checks whether a resolved tenant is active or not. So currently I check for duplicates at that place and throws an exception, which is OK for now. I guess the performance hit of check the tenants for duplicates in each request have to be tolerated.

But I suggest to keep a tab on this issue and let's see how we can handle it better may be in future release?
